internal/v2/s2av2_e2e_test.go

```diff
@@ -455,3 +456,24 @@ func TestNewClientTlsConfigWithoutTokenManager(t *testing.T) {
 		t.Errorf("tls.Config has unexpected certificate: got: %v, want: %v", got, want)
 	}
 }
+
+func TestNewClientTlsConfigWithCustomS2AStream(t *testing.T) {
+	os.Unsetenv(accessTokenEnvVariable)
+	s2aAddr := startFakeS2A(t, "TestNewClientTlsConfig_token")
+	var tokenManager tokenmanager.AccessTokenManager
+	ctx, cancel := context.WithTimeout(context.Background(), defaultE2ETimeout)
+	defer cancel()
+
+	getStreamFuncCalled := false
+	_, err := NewClientTLSConfig(ctx, s2aAddr, nil, tokenManager, s2av2pb.ValidatePeerCertificateChainReq_CONNECT_TO_GOOGLE, "test_server_name", nil, func(ctx context.Context, s2av2Address string) (stream.S2AStream, error) {
+		getStreamFuncCalled = true
+		return s2ATestStream{debug: "test S2A stream"}, nil
+	})
```

Comment:
This is checking the NewClientTLSConfig() func is called, but IIUC we wanted to test if the passed in getStreamFunc is called?

Reply:
It should be testing it underneath: NewClientTLSConfig(..., getStream func param) --> createStream(..., getStream  func param) --> getStream != nil, so invoke.

Can you do a second pass and let me know if you're still seeing this? I might be misunderstanding what you mean.

Reply:
Oh, sry I misread the code, resolving this.